src/l3-resources/atlas-bootstrap/index.ts

```diff
@@ -0,0 +1,129 @@
+   */
+  readonly typesToActivate?: string[];
+}
+
+/**
+ * Generate the CFN extension execution role.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry-public.html
+ */
+export class MongoAtlasBootstrap extends Construct {
+  readonly role: iam.IRole;
+  constructor(scope: Construct, id: string, props?: MongoAtlasBootstrapProps) {
+    super(scope, id);
+
+    this.role = new iam.Role(this, "mongoDBAtlasCFNExecutionRole", {
+      maxSessionDuration: Duration.seconds(8400),
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal("cloudformation.amazonaws.com"),
+        new iam.ServicePrincipal("resources.cloudformation.amazonaws.com"),
+        new iam.ServicePrincipal("lambda.amazonaws.com")
+      ),
+      path: "/",
+      roleName: props?.roleName,
+    });
+
+    const atlasCdkPolicy = new iam.Policy(this, "AtlasCDKExecutionPolicy", {
+      policyName: "AtlasCDKExecution", // Custom Policy Name
+      statements: [
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:DescribeSecret",
+            "secretsmanager:GetSecretValue",
+            "ec2:CreateVpcEndpoint",
+            "ec2:DeleteVpcEndpoints",
+            "cloudformation:CreateResource",
+            "cloudformation:DeleteResource",
+            "cloudformation:GetResource",
+            "cloudformation:GetResourceRequestStatus",
+            "cloudformation:ListResources",
+            "cloudformation:UpdateResource",
+            "iam:AttachRolePolicy",
+            "iam:CreateRole",
+            "iam:DeleteRole",
+            "iam:GetRole",
+            "iam:GetRolePolicy",
+            "iam:ListAttachedRolePolicies",
+            "iam:ListRolePolicies",
+            "iam:PutRolePolicy",
+          ],
+          resources: ["*"],
+        }),
+      ],
+    });
+
+    this.role.attachInlinePolicy(atlasCdkPolicy);
+
+    if (props?.secretProfile) {
+      new MongoSecretProfile(this, "MongoSecretProfile", props?.secretProfile);
+    }
+
+    if (props?.typesToActivate) {
+      for (let type of props?.typesToActivate) {
+        new cloudformation.CfnTypeActivation(this, `${type}TypeActivation`, {
+          executionRoleArn: this.role.roleArn,
+          type: `RESOURCE`,
+          typeName: `MongoDB::Atlas::${type}`,
+          publisherId: "bb989456c78c398a858fef18f2ca1bfc1fbba082",
+        });
+      }
+    }
+  }
+}
+
+export class MongoSecretProfile extends Construct {
+  constructor(scope: Construct, id: string, profileName: string) {
+    super(scope, id);
+
+    // create a secret
+    const secret = new secretsmanager.Secret(this, profileName, {
+      secretName: `cfn/atlas/profile/${profileName}`,
+      description: "Secret used for MongoDB Atlas Cloud Formation api keys",
+      secretStringValue: SecretValue.unsafePlainText(
+        '{"PublicKey":"yourAtlasPublicKey", "PrivateKey": "yourAtlasPrivateKey"}'
```

Comment:
I think we should randomize at least the private key on secret initialization.

```ts
new secretsmanager.Secret(this, id, {
      generateSecretString: {
        secretStringTemplate: JSON.stringify({ PublicKey: 'yourAtlasPublicKey', PrivateKey: '' }),
        generateStringKey: 'PrivateKey',
        excludeCharacters: '%+~`#$&*()|[]{}:;<>?!\'/@"\\=-.,',
      },
    });
```

And you got the initial secret like this:

<img width="820" alt="image" src="https://github.com/mongodb/awscdk-resources-mongodbatlas/assets/278432/f32f771a-7b38-40e1-bb59-18417505b3a9">

Thoughts?